api/schedule/create.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/go-vela/server/database"
+	"github.com/go-vela/server/router/middleware/repo"
+	"github.com/go-vela/server/router/middleware/user"
+	"github.com/go-vela/server/util"
+	"github.com/go-vela/types/library"
+	"github.com/sirupsen/logrus"
+)
+
+// swagger:operation POST /api/v1/schedules/{org}/{repo} schedules CreateSchedule
+//
+// Create a schedule in the configured backend
+//
+// ---
+// produces:
+// - application/json
+// parameters:
+// - in: path
+//   name: org
+//   description: Name of the org
+//   required: true
+//   type: string
+// - in: path
+//   name: repo
+//   description: Name of the repo
+//   required: true
+//   type: string
+// - in: body
+//   name: body
+//   description: Payload containing the schedule to create
+//   required: true
+//   schema:
+//     "$ref": "#/definitions/Schedule"
+// security:
+//   - ApiKeyAuth: []
+// responses:
+//   '201':
+//     description: Successfully created the schedule
+//     schema:
+//       "$ref": "#/definitions/Schedule"
+//   '400':
+//     description: Unable to create the schedule
+//     schema:
+//       "$ref": "#/definitions/Error"
+//   '403':
+//     description: Unable to create the schedule
+//     schema:
+//       "$ref": "#/definitions/Error"
+//   '409':
+//     description: Unable to create the schedule
+//     schema:
+//       "$ref": "#/definitions/Error"
+//   '500':
+//     description: Unable to create the schedule
+//     schema:
+//       "$ref": "#/definitions/Error"
+//   '503':
+//     description: Unable to create the schedule
+//     schema:
+//       "$ref": "#/definitions/Error"
+
+// CreateSchedule represents the API handler to
+// create a schedule in the configured backend.
+func CreateSchedule(c *gin.Context) {
+	// capture middleware values
+	u := user.Retrieve(c)
+	r := repo.Retrieve(c)
+	allowlist := c.Value("allowlistschedule").([]string)
+	minimumFrequency := c.Value("scheduleminimumfrequency").(time.Duration)
+
+	// capture body from API request
+	input := new(library.Schedule)
+
+	err := c.Bind(input)
+	if err != nil {
+		retErr := fmt.Errorf("unable to decode JSON for new schedule: %w", err)
+
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+
+	// ensure the entry is valid
+	err = validateEntry(minimumFrequency, input.GetEntry())
+	if err != nil {
+		retErr := fmt.Errorf("schedule of %s is invalid: %w", input.GetName(), err)
+
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+
+	// ensure schedule name is defined
+	if input.GetName() == "" {
+		util.HandleError(c, http.StatusBadRequest, fmt.Errorf("schedule name must be set"))
+	}
+
+	// update engine logger with API metadata
+	//
+	// https://pkg.go.dev/github.com/sirupsen/logrus?tab=doc#Entry.WithFields
+	logrus.WithFields(logrus.Fields{
+		"org":  r.GetOrg(),
+		"repo": r.GetName(),
+		"user": u.GetName(),
+	}).Infof("creating new schedule %s", input.GetName())
+
+	// ensure repo is allowed to be activated
```

Comment:
Thoughts on updating the wording here slightly?

```suggestion
	// ensure repo is allowed to create new schedules
```